builtin/blame.c

```diff
@@ -902,6 +902,11 @@ int cmd_blame(int argc, const char **argv, const char *prefix)
 	long anchor;
 	const int hexsz = the_hash_algo->hexsz;
 
+	if (startup_info->have_repository) { /* no repo when running with -h */
```

Comment:
A third option! (Maybe `nongit` was specific to `diff`?)

Reply:
If this is just for `-h`, should it be moved further down the method to be after the `-h` is complete? Is that possible?

Reply:
Nongit was not available in blame. I decided to add this check explicitly to make the scenario I'm guarding against as clear as possible. However, if moving down would align better with other commands we've adapted for sparse index I'm happy to do that instead (and will probably try to follow suit in `pack-objects` and `checkout`.

Reply:
The location isn't _terribly_ important. But I do think the `pack-objects` and `checkout` cases could have the same `if (startup_info->have_repository)` check.

Reply:
Sounds good.